examples/test/videosphere/hls.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Videosphere HLS</title>
+    <meta name="description" content="Videosphere HLS - A-Frame">
+    <script src="../../../dist/aframe-master.js"></script>
+  </head>
+  <body>
+    <a-scene stats>
+      <a-assets>
+        <video id="city" preload="true" webkit-playsinline playsinline
+               src="../../assets/video/city_halfres_hls.m3u8"
```

Comment:
So, this video plays properly on my **iPhone 5** running **10.3.2 (Safari/602.1)** (Safari/604.1), but the video fails to be displayed on my **iPhone 7** running **iOS 11.0.2** (Safari/604.1). (But in my Remote Debugging, `console.log` seems to say the video is in fact playing and emitting events.)

I tried adding the `muted` attribute hoping it'd fix the [complex autoplay restrictions in iOS 10](https://webkit.org/blog/6784/new-video-policies-for-ios/), but that didn't seem to do anything.

If I comment out all the A-Frame bits and just have the `<video>` play on the page, the HLS video plays just fine (in both versions of iOS). I'm not entirely sure, but my hunch is that this is related to the `<video>` not being visible on-screen (per the aforementioned [iOS 10+ video policies](https://webkit.org/blog/6784/new-video-policies-for-ios/)).

I tried a few other things, such as changing `scripts/budo.js` to serve the `.m3u8` files with a different response (`Content-Type: application/x-mpegURL` instead of `Content-Type: application/vnd.apple.mpegurl`). But that didn't seem to do anything.

<hr>

FWIW, I was able to get a non-HLS, `.mp4`, video to play. (I tested with the original `.mp4` and also with a separate version converted from the `.m3u8` without any audio tracks – still no luck.)

<hr>

it seems other folks are encountering the same issue in the [`video.js`](https://github.com/videojs/video.js) project: https://github.com/videojs/video.js/issues/4296#issuecomment-314792383

which version of iOS are you testing this on? I'm getting real curious what the problem is. we all might be running into a regression in Safari/WebKit on iOS 10.3.x+ (but works in 10.3.1 on my iPhone 5 device).

Reply:
11+ -- I had tested it to work properly on iOS 10.  I suspect a regression in iOS 11

Reply:
ah, right so you had it working on it 10.3.3. so yeah must’ve happened during a patch, minor, or major version change.

wonder if with the video.js community and reaching out to the WebKit authors if we could track this down. perhaps reaching out to the author of that WebKit video policies blog post might be a good start.

Reply:
videos it is

Reply:
The MP4 path is separate from the HLS path, and (currently) works without quirks when it works.